cli/repl.rs

```diff
@@ -29,17 +32,64 @@ impl Validator for Helper {
   }
 }
 
+async fn post_message_and_poll(
+  worker: &mut Worker,
+  session: &mut InspectorSession,
+  method: &str,
+  params: Option<Value>,
+) -> Result<Value, AnyError> {
+  let response = session.post_message(method.to_string(), params);
+  tokio::pin!(response);
+
+  loop {
+    tokio::select! {
+      result = &mut response => {
+        return result
+      }
+
+      _ = &mut *worker => {
+        tokio::time::delay_for(tokio::time::Duration::from_millis(0)).await;
+      }
+    }
+  }
+}
+
+async fn read_line_and_poll(
+  worker: &mut Worker,
+  editor: Arc<Mutex<Editor<Helper>>>,
+) -> Result<String, ReadlineError> {
+  let mut line =
+    tokio::task::spawn_blocking(move || editor.lock().unwrap().readline("> "));
+
+  loop {
+    tokio::select! {
+      result = &mut line => {
+        return result.unwrap();
+      }
+      _ = &mut *worker => {
+        tokio::time::delay_for(tokio::time::Duration::from_millis(0)).await;
```

Comment:
Just returning `line.await` worked great locally but completely locks up the CI with no test output at all.